main.py

```diff
@@ -829,6 +833,64 @@ def reportimage():
 
     return better_jsonify(**resp)
 
+@app.route("/suggest", methods=["GET"])
+@cache.cached(timeout=300, key_prefix='suggestions', query_string=True)
+def suggest(limit=10):
+    """ Return suggestions for query field autocompletion """
+    limit = request.args.get('limit', limit)
+    txt = request.args.get("q", "").strip()
+
+    suggestions = list()
+    whois_prefix = "Hver er "
+    whatis_prefix = "Hvað er "
+
+    prefix = None
+    if txt.startswith(whois_prefix):
+        prefix = whois_prefix
+    elif txt.startswith(whatis_prefix):
+        prefix = whatis_prefix
+
+    if not txt or not prefix:
+        return better_jsonify(suggestions=suggestions)
+
+    with SessionContext(commit=False) as session:
+        name = txt[len(prefix):].strip()
+
+        # Hver er Jón Jónsson ?
+        if prefix is whois_prefix and name[0].isupper():
+            # Find matching persons ordered by number of 
+            # related articles over the past year
+            q = (session.query(Person.name, dbfunc.count(Article.id).label('total'))
+                .filter(Person.name.ilike(name + '%'))
+                .join(Article)
+                .filter(Article.timestamp > datetime.utcnow() - timedelta(days=365))
+                .group_by(Person.name)
```

Comment:
Don't you need to group on a sum of total?

Reply:
Maybe I'm just dim-witted today, but I don't see the problem with the query. The SQL generated by SQLAlchemy is "SELECT persons.name AS persons_name, count(articles.id) AS total FROM persons JOIN articles ON articles.url = persons.article_url GROUP BY persons.name ORDER BY total DESC;"

Isn't that exactly what we want?